ign_ros2_control/src/ign_system.cpp

```diff
@@ -642,10 +663,12 @@ hardware_interface::return_type IgnitionSystem::write(
         this->dataPtr->ecm->CreateComponent(
           this->dataPtr->joints_[i].sim_joint,
           ignition::gazebo::components::JointVelocityCmd({target_vel}));
-      } else if (!vel->Data().empty()) {
-        vel->Data()[0] = target_vel;
-      } else if (!vel->Data().empty()) {
-        vel->Data()[0] = target_vel;
```

Comment:
Whoops, I duplicated this block of code. Plus, I think it is safer to use:

```
        const auto jointVelCmd =
          this->dataPtr->ecm->Component<ignition::gazebo::components::JointVelocityCmd>(
          this->dataPtr->joints_[i].sim_joint);
        *jointVelCmd = ignition::gazebo::components::JointVelocityCmd(
          {target_vel});
```

Than...

`vel->Data()[0] = target_vel;`

I might be wrong, though.